WooCommerce/Classes/ViewRelated/Orders/Order Details/Shipping Labels/Create Shipping Label Form/Package Details/Package Selection/Package Creation/ShippingLabelServicePackageList.swift

```diff
@@ -0,0 +1,73 @@
+import SwiftUI
+import Yosemite
+
+struct ShippingLabelServicePackageList: View {
+    @Environment(\.presentationMode) var presentation
+    @StateObject var viewModel = ShippingLabelServicePackageListViewModel()
+    let packagesResponse: ShippingLabelPackagesResponse?
+    let safeAreaInsets: EdgeInsets
+
+    var body: some View {
+        LazyVStack(spacing: 0) {
+            ListHeaderView(text: Localization.servicePackageHeader, alignment: .left)
+                .padding(.horizontal, insets: safeAreaInsets)
+
+            /// Packages
+            ///
+            ForEach(viewModel.predefinedOptions, id: \.title) { option in
+
+                ListHeaderView(text: option.title.uppercased(), alignment: .left)
+                    .padding(.horizontal, insets: safeAreaInsets)
+                ForEach(option.predefinedPackages) { package in
+                    let selected = package == viewModel.selectedPackage
+                    SelectableItemRow(title: package.title,
+                                      subtitle: package.dimensions + " \(viewModel.dimensionUnit)",
+                                      selected: selected)
+                        .onTapGesture {
+                            viewModel.selectedPackage = package
+                        }
+                        .padding(.horizontal, insets: safeAreaInsets)
+                        .background(Color(.systemBackground))
+                    Divider().padding(.leading, Constants.horizontalPadding)
+                }
+            }
+        }
+        .onAppear(perform: {
+            viewModel.packagesResponse = packagesResponse
+        })
```

Comment:
If I understand correctly, this is for demo purpose only right? Since updating the view model `onAppear` is causing the view to be redrawn every time switching back and forth between the two tabs of the Add New Package screen, which results in losing selection state of the list. But please ignore this comment if you intend to remove this part when the package response is fetched from remote.

Reply:
I used `onAppear` here as a way to [add values to view model properties when initializing the view model as a `StateObject`](https://sarunw.com/posts/how-to-initialize-stateobject-with-parameters-in-swiftui/). I was planning to keep using this approach with the real data, because I wasn't thinking about trying to retain each tab's state when switching between tabs.

I'll have to think this over — I guess we'll probably have to initialize the view models for the tabs in the "Add New Package" parent view, so I'll work on that in my next PR.